seisbench/models/dkpn.py

```diff
@@ -0,0 +1,784 @@
+
+            self.down_branch.append(nn.ModuleList([conv_same, bn1, conv_down, bn2]))
+
+        for i in range(self.depth - 1):
+            filters = int(2 ** (3 - i) * self.filters_root)
+            conv_up = nn.ConvTranspose1d(
+                last_filters, filters, self.kernel_size, self.stride, bias=False
+            )
+            last_filters = filters
+            bn1 = nn.BatchNorm1d(filters, eps=1e-3)
+            conv_same = nn.Conv1d(
+                2 * filters, filters, self.kernel_size, padding="same", bias=False
+            )
+            bn2 = nn.BatchNorm1d(filters, eps=1e-3)
+
+            self.up_branch.append(nn.ModuleList([conv_up, bn1, conv_same, bn2]))
+
+        self.out = nn.Conv1d(last_filters, self.classes, 1, padding="same")
+        self.softmax = torch.nn.Softmax(dim=1)
+
+        # DKPN attributes for taking care of windows CF and probs
+        self.batch_ = []
+        self.batch_cfs = []
+        self.batch_probs = []
+        self.stream_cfs = None
+
+    def __reset_predict(self):
+        self.batch_ = []
+        self.batch_cfs = []
+        self.batch_probs = []
+        self.stream_cfs = None
+
+    def forward(self, x, logits=False):
+        x = self.activation(self.in_bn(self.inc(x)))
+
+        skips = []
+        for i, (conv_same, bn1, conv_down, bn2) in enumerate(self.down_branch):
+            x = self.activation(bn1(conv_same(x)))
+
+            if conv_down is not None:
+                skips.append(x)
+                if i == 1:
+                    x = F.pad(x, (2, 3), "constant", 0)
+                elif i == 2:
+                    x = F.pad(x, (1, 3), "constant", 0)
+                elif i == 3:
+                    x = F.pad(x, (2, 3), "constant", 0)
+
+                x = self.activation(bn2(conv_down(x)))
+
+        for i, ((conv_up, bn1, conv_same, bn2), skip) in enumerate(
+            zip(self.up_branch, skips[::-1])
+        ):
+            x = self.activation(bn1(conv_up(x)))
+            x = x[:, :, 1:-2]
+
+            x = self._merge_skip(skip, x)
+            x = self.activation(bn2(conv_same(x)))
+
+        x = self.out(x)
+        if logits:
+            return x
+        else:
+            return self.softmax(x)
+
+    @staticmethod
+    def _merge_skip(skip, x):
+        offset = (x.shape[-1] - skip.shape[-1]) // 2
+        x_resize = x[:, :, offset : offset + skip.shape[-1]]
+
+        return torch.cat([skip, x_resize], dim=1)
+
+    # ================================================================
+    # ==================================  PROCESS STREAM
+    # ================================================================
+
+    def annotate_stream_pre(self, stream, argdict):
+        """ ARGDICT represent the default dict,merged (eventually)
+            with the key:value pairs of the "*.json" specifying the model
+        """
+        self.__reset_predict()
+
+        # -----------------   Pre-pend the original code of SEISBENCH (v0.2)
```

Comment:
I think you can just call `stream = super().annotate_stream_pre(stream, argdict)` instead.